plugins/widget/plugin.js

```diff
@@ -3256,11 +3485,15 @@
 
 			for ( cl in previousClasses ) {
 				// Avoid removing and adding classes again.
-				if ( !( newClasses && newClasses[ cl ] ) )
+				if ( !( newClasses && newClasses[ cl ] ) ) {
 					this.removeClass( cl );
+					addRemoveClassToStyleDef( this, cl, 1 );
```

Comment:
Shouldn't it be moved into `removeClass`?

Reply:
I'm reverting this change because of 

>Easy Image has the same issue as image2: data.classes in newly created widget does not contain .easyimage class, yet styleDefinition – does. Maybe such behavior is hardcoded into widget plugin itself?

`styleDefinition.attributes.class` should be updated on `data`, not on every call of `addClass` or `removeClass`